### PR TITLE
Added import_keypair method for DigitalOcean

### PR DESCRIPTION
**Pull request by one-hero**
### What does this PR do?

Adds capability for `import_keypair` method to be also used for DigitalOcean
### What issues does this PR fix or reference?

PR for #32568
### Tests written?

No

### CODE REVIEW

File: salt/cloud/clouds/digital_ocean.py [hunk 1]
```diff
@@ -663,6 +663,27 @@ def show_keypair(kwargs=None, call=None):
     return details
 
 
+def import_keypair(kwargs=None, call=None):
+    '''
+    Upload public key to cloud provider.
+    Similar to EC2 import_keypair.
+
```
Comment (rallytime) [inline salt/cloud/clouds/digital_ocean.py]: Can you add a `.. versionadded:: Carbon` tag here?

File: salt/cloud/clouds/digital_ocean.py [hunk 2]
```diff
@@ -663,6 +663,27 @@ def show_keypair(kwargs=None, call=None):
...
+    kwargs
+        file(mandatory): public key file-name
+        keyname(mandatory): public key name in the provider
+    '''
+    with open(kwargs['file'], 'r') as public_key_filename:
```
Comment (rallytime) [inline salt/cloud/clouds/digital_ocean.py]: Can you please change the `open` call to use Salt's `fopen` function instead? See [here](https://github.com/saltstack/salt/blob/b6d640da0db7a5a57709b48d8dc1d19509901e32/salt/utils/__init__.py#L1182) for reference. The prevents leaking file descriptors.

**Comment by rallytime** [comment]
@one-hero Thank you for this addition! I left a couple of small comments, but once those are cleaned up, we'd be glad to get this in.

**Comment by one-hero** [comment]
@rallytime I've updated the PR. 
Please take another look, thanks

**Comment by rallytime** [comment]
@one-hero Great! Thank you!

**Pull request merged**